tests/boot/boot_from_pxe.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+use strict;
+use warnings;
+use File::Basename;
+use base "opensusebasetest";
+use testapi;
+
+sub run() {
+    check_screen("virttest-bootloader", 60);
+    my $image_path = "";
+
+    #detect pxe location
+    if (match_has_tag("virttest-bootloader")) {
+        #BeiJing
+        # Wait the second screen for ipmi bootloader
+        send_key_until_needlematch "virttest-boot-into-pxe", "f12", 3, 60;
+
+        # Wait pxe management screen
+        send_key_until_needlematch "virttest-pxe-menu", "f12", 200, 1;
+
+        # Login to command line of pxe management
+        send_key_until_needlematch "virttest-pxe-edit-prompt", "esc", 60, 1;
+
+        $image_path = get_var("HOST_IMG_URL");
+    }
+    else {
+        #Numburg
+        if (!match_has_tag("qa-net-selection")) {
+            assert_screen "qa-net-selection", 240;
+        }
+
+        #send_key_until_needlematch "qa-net-selection-" . get_var('DISTRI') . "-" . get_var("VERSION"), 'down', 30, 3;
+        #Don't use send_key_until_needlematch to pick first menu tier as dist network sources might not be ready when openQA is running tests
+        send_key 'esc';
+        assert_screen 'qa-net-boot';
+
+        my $image_name = "";
+        if (check_var("INSTALL_TO_OTHERS", 1)) {
+            $image_name = get_var("REPO_0_TO_INSTALL");
+        }
+        else {
+            $image_name = get_var("REPO_0");
+        }
+
+        my $arch = get_var('ARCH');
+        my $path = "/mnt/openqa/repo/${image_name}/boot/${arch}/loader";
+        my $repo = get_var('HOST') . "/assets/repo/${image_name}";
+        $image_path = "$path/linux initrd=$path/initrd install=$repo";
+    }
+
+    my $type_speed = 20;
```

Comment:
There is a function `type_string_slow` in lib/utils.pm. Would this work? If you have a good reason why 20 is better this is also ok.